crates/sui/src/client_commands.rs

```diff
@@ -1173,18 +1266,57 @@ impl SuiClientCommands {
                 type_args,
                 gas,
                 gas_budget,
+                dry_run,
                 gas_price,
                 args,
                 serialize_unsigned_transaction,
                 serialize_signed_transaction,
             } => {
-                let tx_data = construct_move_call_transaction(
-                    package, &module, &function, type_args, gas, gas_budget, gas_price, args,
-                    context,
+                let tx_kind = construct_move_call_transaction_kind(
+                    package, &module, &function, type_args, args, context,
                 )
                 .await?;
+                let gas_owner = context.try_get_object_owner(&gas).await?;
+                let signer = gas_owner.unwrap_or(context.active_address()?);
+                let gas_price = gas_price.unwrap_or(context.get_reference_gas_price().await?);
+
+                let client = context.get_client().await?;
+                if dry_run {
+                    return helper_execute_dry_run(
+                        context,
+                        signer,
+                        tx_kind,
+                        gas_budget,
+                        gas_price,
+                        gas.map(|x| vec![x]),
+                        None,
+                    )
+                    .await;
+                }
+                // This is what it's being doen in the original move_call code,
+                // and it is passed to the select_gas function.
```

Comment:
nit: typos in comment

Reply:
Can we generalise the implementation of `tx_data` so that it extracts the input objects from `tx_kind` itself? It seems like that should be doable.

Reply:
Not sure I follow here. In this one place (`Call` command), the `input_coins` are taken out of the tx_kind, but in other commands, `input_coins` are passed as arguments.

Reply:
Oh, now I see, I think we can do that.

Reply:
I would do that though in a separate PR. I am not confident about one part of it.